core/state/statedb.go

```diff
@@ -1450,9 +1532,17 @@ func (s *StateDB) Commit(failPostCommitFunc func(), postCommitFuncs ...func() er
 				}
 				// Only update if there's a state transition (skip empty Clique blocks)
 				if parent := s.snap.Root(); parent != s.expectedRoot {
-					if err := s.snaps.Update(s.expectedRoot, parent, s.snapDestructs, s.snapAccounts, s.snapStorage, verified); err != nil {
+					snapAccountLock.Lock()
```

Comment:
Be noted `if s.snap != nil` in line 1526 - if the snap is broken, the original pipecommit will also not as expected, for no snap will be generated ever ( only restart with clean data0.